README.md

```diff
@@ -50,6 +50,115 @@ data.
 
 ## Implementation details
 
+### ept.json
+
+This file shall follow the format of EPT specified [here](https://entwine.io/entwine-point-tile.html#ept-json). The `dataType` parameter must be set to `copc` and the `version` parameter must be at least `"2.0.0.0"`.
+
+### ept-data
+
+This folder shall contain a single file named `octree.laz`. This file must abide by the LAZ 1.4 format. Each "node" of the octree shall be stored in the LAZ file as its own compressed chunk (see [here](https://www.cs.unc.edu/~isenburg/lastools/download/laszip.pdf)), with chunks and points within the chunk having no meaningful ordering unless otherwise defined.
+
+### ept-hierarchy
+
+This folder shall contain one or more json files detailing information about the octree nodes, where string keys of `D-X-Y-Z` map to an object. For example, for the root octree node `ept-heirarchy/0-0-0-0.json`:
+
+```json
+{
+  "0-0-0-0": {
+    point_count: 65341,
+    chunk_size: 6000,
+    chunk_offset: 0
+  },
+  "1-0-0-0": {
+    point_count: 438,
+    chunk_size: 40,
+    chunk_offset: 6000
+  },
+  "2-0-1-0": {
+    point_count: 322,
+    chunk_size: 30,
+    chunk_offset: 6040
+  },
+  "2-0-1-2": {
+    point_count: 4332,
+    chunk_size: 400,
+    chunk_offset: 6070
+  },
+  "1-0-0-1": {
+    point_count: 56209,
+    chunk_size: 5000,
+    chunk_offset: 11070
+  },
+  "3-0-0-0": {},
+    ...
+}
+```
+
+#### `point_count`
+
+This integer attribute defines the total number of points stored in this node.
+
+#### `chunk_offset`
+
+This integer value defines the starting byte within the LAZ file for this chunk, relative to the beginning of point data. Therefore, for the first chunk in the LAZ file, the absolute starting byte location within the file will be calculated from `(header: offset to point data) + (chunk_offset) = file offset`, so within the json file, the first chunk will always have a value of `0`.
+
+Note that this value **must** be aligned correctly with the LAZ's internal chunk table. Therefore, removing or adding points within the `octree.laz` file is illegal.
+
+#### `chunk_size`
+
+This value is the byte length of the current chunk relative to `chunk_offset`. Therefore, for any given node at position `i`, 
+
+```
+chunk_offset[i] = chunk_offset[i-1] + chunk_size[i-1]
+file_offset[i] = chunk_offset[i] + header.offset_to_point_data
+```
+
+Each hierarchy node must contain at least the above attributes, except for an empty object `{}` which indicates this node's metadata and its subtree resides in its octree file. For example, in the above example, there must be an additional file `ept-hierarchy/3-0-0-0.json`:
```

Comment:
I think the value should be `null` or  `-1`, an empty object might get weird to check existence of...

```suggestion
Each hierarchy node must contain at least the above attributes, except for `null` which indicates this node's metadata and its subtree resides in a separate file. For example, with `"3-0-0-0":  null` there must be an additional file `ept-hierarchy/3-0-0-0.json`:
```

Reply:
> I think the value should be `null` or `-1`, an empty object might get weird to check existence of...

-1 would keep it somewhat consistent with standard EPT.